File: implementations/vb/Loom.Core/Models/RagContext.vb
Namespace Models
    ''' <summary>
    ''' All chunks recovered and the chosen strategy
    ''' </summary>
    Public Class RagContext
        Public Property Strategy As Enums.InjectionStrategy = Enums.InjectionStrategy.Sectioned
        Public Property MaxTokens As Integer = 2000 ' recovered by retriever
        Public Property RetrievedChunks As New List(Of RagChunk)
    End Class
End Namespace

File: implementations/csharp/src/Loom.Core/Models/RagContext.cs
using System.Collections.Generic;

namespace Loom.Core.Models
{
    /// <summary>
    /// All chunks recovered and the chosen strategy
    /// </summary>
    public class RagContext
    {
        public int MaxTokens { get; set; } = 2000; // recovered by retriever
        public Enums.InjectionStrategy Strategy { get; set; }
        public List<RagChunk> RetrievedChunks { get; set; } = new List<RagChunk>();

    }
}
